shared/Microsoft.Extensions.Buffers.Sources/MemoryPoolSlab.cs

```diff
@@ -66,15 +76,36 @@ protected void Dispose(bool disposing)
 
                 _isActive = false;
 
+                _disposedValue = true;
+            }
+        }
+
+        internal void Return(MemoryPoolBlock block)
+        {
+#if BLOCK_LEASE_TRACKING
+            Debug.Assert(block.Slab == this, "Returned block was not this slab");
+            Debug.Assert(block.IsLeased, $"Block being returned to pool twice: {block.Leaser}{Environment.NewLine}");
+            block.IsLeased = false;
+#endif
+            GC.SuppressFinalize(block);
+
+            var remaining = Interlocked.Decrement(ref _ownedBlocks);
+
+            if (remaining < 0)
```

Comment:
This case is double return. To check for actual dispose use `_disposedValue`. Can we throw only when the block we are turning we double returned?

Reply:
We do with assert. Should we change assert to exception? What's easier to debug?